platform/core/src/services/MeasurementService/MeasurementService.test.js

```diff
@@ -0,0 +1,391 @@
+import MeasurementService from './MeasurementService.js';
+import log from '../../log';
+
+jest.mock('../../log.js', () => ({
+  info: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn(),
+}));
+
+describe('MeasurementService.js', () => {
+  let measurementService;
+  let measurement;
+  let source;
+  let definition;
+  let matchingCriteria;
+  let toAnnotation;
+  let toMeasurement;
+  let annotation;
+
+  beforeEach(() => {
+    measurementService = new MeasurementService();
+    source = measurementService.createSource('Test', '1');
+    definition = 'Length';
+    annotation = {
+      toolName: definition,
+      measurementData: {},
+    };
+    measurement = {
+      sopInstanceUID: '123',
+      frameOfReferenceUID: '1234',
+      referenceSeriesUID: '12345',
+      label: 'Label',
+      description: 'Description',
+      unit: 'mm',
+      area: 123,
+      type: measurementService.VALUE_TYPES.POLYLINE,
+      points: [{ x: 1, y: 2 }, { x: 1, y: 2 }],
+      source: source,
+    };
+    toAnnotation = () => annotation;
+    toMeasurement = () => measurement;
+    matchingCriteria = {
+      valueType: measurementService.VALUE_TYPES.POLYLINE,
+      points: 2,
+    };
+    log.warn.mockClear();
+    jest.clearAllMocks();
+  });
+
+  describe('createSource()', () => {
+    it('creates new source', () => {
+      measurementService.createSource('Testing', '1');
+    });
+
+    it('returns warning if no name provided', () => {
+      measurementService.createSource(null, '1');
+
+      expect(log.warn.mock.calls.length).toBe(1);
+    });
+
+    it('returns warning if no version provided', () => {
+      measurementService.createSource('Testing', null);
+
+      expect(log.warn.mock.calls.length).toBe(1);
+    });
+  });
+
+  describe('addMapping()', () => {
+    it('adds new mapping', () => {
+      measurementService.addMapping(
+        source,
+        definition,
+        matchingCriteria,
+        toAnnotation,
+        toMeasurement
+      );
```

Comment:
Is there a way for us to assert that the mapping was added? Do we expose mapping in any way from the `MeasurementSource`? Maybe be testing that a very simple mapping is used when retrieving a specific source's measurements?

Reply:
We can have a getter for mappings like getMappings our have mappings accessible through source, not sure if this is something we would need

About the test, there is one test which im adding a simple mapping to test getAnnotation functionality